lib/components/beta-bar/index.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import SmallCrossIcon from '../../icons/cross-small';
+import { isElectron } from '../../utils/platform';
+
+const loadOptOut = (): boolean => {
+  try {
+    const stored = localStorage.getItem('betaBannerDismissedAt');
+    const dismissedAt = stored ? parseInt(stored, 10) : -Infinity;
+
+    return (
+      isNaN(dismissedAt) || Date.now() - dismissedAt > 7 * 24 * 60 * 60 * 1000
+    );
+  } catch (e) {
+    return true;
+  }
+};
+
+const storeOptOut = () => {
+  try {
+    localStorage.setItem('betaBannerDismissedAt', Date.now().toString());
+  } catch (e) {
+    // pass
+  }
+};
+
+const BetaBar = () => {
+  const [isVisible, setIsVisible] = useState(loadOptOut());
+  const dismissBanner = () => {
+    storeOptOut(), setIsVisible(false);
```

Comment:
you've probably already spotted this, and while I love the comma operator and originally doodled out a solution using it, it's probably a typo here

we _could_ do `() => (storeOptOut(), setIsVisible(false))` which would be more normative for comma operator expressions, but within the `{}`s it's awkward; probably best separated by semi-colons `() => { storeOptOut(); setIsVisible(false); }`

Reply:
I say "normative" because comma-operator expressions return the last value in their chain but when put in `{}`s without a return statement nothing is returned, meaning we're using a construct built to sequence operations and return the last value but intentionally ignoring that value then not returning anything

Reply:
noooope that was a typo and yet somehow worked